optaplanner-core/src/main/java/org/optaplanner/core/impl/constructionheuristic/placer/QueuedValuePlacerFactory.java

```diff
@@ -53,7 +53,7 @@ public QueuedValuePlacerFactory(QueuedValuePlacerConfig placerConfig) {
         ValueSelectorConfig valueSelectorConfig_ = buildValueSelectorConfig(configPolicy, entityDescriptor);
         ValueSelector<Solution_> valueSelector = ValueSelectorFactory.<Solution_> create(valueSelectorConfig_)
                 .buildValueSelector(configPolicy, entityDescriptor, SelectionCacheType.PHASE, SelectionOrder.ORIGINAL,
-                        false);
+                        false, true);
```

Comment:
I hate to do this.

Reply:
Added a couple of tests for bugs that I ran into during development. With the extra coverage I can look if this can be refactored without the fear of breaking something.

Reply:
Not going to improve this in this PR.

# How filtering works

The `false, true` flags are overrides of the reinitializable/reassignable filtering behavior that is an optional feature of the **ValueSelector** (keep this in mind).

The reinitializable filtering is specified by the `configPolicy`, which is an approach that covers most situations.

The reassign filtering is only used by the list variable and the Allocate to value from queue algorithm so there's not much point in adding it to the `configPolicy`. This line is the only place where the reassignable filtering needs to be enabled or disabled.

## 1. Local search
In the Local Search phase, reinitializable/reassignable filtering is not needed because the solution is already initialized. The configPolicy doesn't enable it and its setting is honored.

## 2. Construction heuristics
In the Consturction Heuristic (and Exhaustive Search) phase this type of filtering is needed so that only uninitialized entities and unassigned values are touched. The configPolicy enables the filtering but in some cases (covered by this line) it cannot be honored. See below.

### 2.1. Allocate entity from queue
The Allocate entity from queue algorithm (including the six specialized "Fit" algorithms) uses a single ValueSelector. The configPolicy enables the filtering and the setting is honored. This is the simple case.

### 2.2. Allocate to value from queue
The Allocate to value from queue algorithm uses ValueSelector mimicking so it has two ValueSelectors. Only one of them must use the filtering.

#### 2.2.1. Basic variable
The recording (leading) ValueSelector doesn't use filtering, so the configPolicy setting is being overridden here using the `false` flag. The replaying ValueSelector honors the configPolicy setting and uses reinitializable filtering.

#### 2.2.2. List variable
The recording ValueSelector uses reassignable filtering to avoid selecting values that are already assigned. The filtering is enabled here by the `true` flag. Otherwise, it's not used. That includes the replaying ValueSelector and any ValueSelectors created in the Local Search phase.

# Conclusion
The decision whether to use which filtering is rather complex and is quiet difficult to track down. The code doesn't make it apparent. If the situation can be improved, it won't be without substantial refactoring. This is out of scope of this PR.

Reply:
TODO add a TODO.